content/contribute/contributing-and-managing-patterns.adoc

```diff
@@ -0,0 +1,145 @@
+
+For example:
+
+[source,terminal]
+----
+$ common/scripts/make_common_subtree.sh https://github.com/mhjacks/common.git wip-main common-subtree
+----
+
+This replaces the common directory in the current repository with the wip-main branch from mhjacks's common repository and names the remote `common-subtree`.
+
+To pull changes from mhjacks's wip-main branch into the parent repository:
+
+[source,terminal]
+----
+$ git subtree pull --prefix common common-subtree wip-main
+----
+
+Running the script without arguments defaults to:
+
+[source,terminal]
+----
+$ common/scripts/make_common_subtree.sh https://github.com/validatedpatterns/common.git main common-subtree
+----
+
+These are the default settings for the repository configuration.
+
+[id="subtree-vs-submodule"]
+== Subtree vs. Submodule
+
+Ensuring patterns can be easily shared across multiple projects has always been a priority. While it is technically possible to share changes between unrelated Git repositories, this is a largely manual process prone to errors. We aim to offer a seamless "pull" experience, allowing a single `git pull` action to update shared pattern components. For repository sharing, we considered two main strategies: submodules and subtrees. Initially, we used submodules but later moved to subtrees.
+
+Subtrees integrate the history of another repository into a parent repository, offering many benefits of submodules without most of their drawbacks. Atlassian provides useful documentation on subtrees, which explains their advantages in more detail. For more information see, link:https://www.atlassian.com/blog/developer[Developer's blog] and link:https://www.atlassian.com/git/tutorials/git-subtree[Git subtree]. 
+
+Submodules presented unavoidable challenges, such as:
+
+* Remembering to check out repositories with `--recurse-submodules` or running `git submodule init && git submodule sync`. The common directory often appeared empty without these steps.
+* Ensuring compatibility between submodule-based repositories and other tools. While tools like ArgoCD and Tekton Pipelines supported submodules well, compatibility concerns remained.
+* Switching branches where different revisions of submodules were referenced often resulted in out-of-sync states. This behavior, although is correct, could be confusing.
+* Submodules required mirroring more repositories in disconnected environments.
+* Working with forked submodules meant maintaining two separate repositories and multiple branches in each.
+
+Subtrees also have challenges. For example, it is easier to diverge from the upstream version of the subtree, and users may not realize that a subtree is in use when cloning a repository. This can be considered a feature, but may lead to conflicts when updating to a newer version. Additionally, subtrees are less commonly understood, which can result in unexpected issues, such as:
```

Comment:
Consider using 'might' or  'can' rather than 'may' unless                            updating existing content that uses the term.